src/context-selection/org-unit-selector-bar-item/use-org-unit-paths-by-name.js

```diff
@@ -11,19 +11,14 @@ export default function useOrgUnitPathsByName(searchTerm) {
             },
         },
     ]
-    const {
-        isIdle,
-        isLoading: loading,
-        error,
-        data,
-    } = useQuery(queryKey, {
+    const { isFetched, isFetching, error, data } = useQuery(queryKey, {
         enabled: !!searchTerm,
         select: (data) => data.organisationUnits.map(({ path }) => path),
     })
 
     return {
-        called: !isIdle,
-        loading,
+        called: isFetched,
+        loading: isFetching,
```

Comment:
I think our usual strategy is as you described -- I'll check this out to see if there's a reason to use `isFetching`, but I'll ping @Birkbjo for his reasoning

Reply:
So this was done to fix issues that occurred after the update to react query. So the semantics are still kind of similar to V3, but there's now two different states to represent the query-state (see https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#the-idle-state-has-been-removed). This affected queries that are disabled by default, eg. `enabled: !!searchTerm` here. The difference is that `status` will be `loading`, and `fetchStatus: 'idle'` (as opposed to `status: 'idle'` in V3), and it seems that `isLoading` is still just reflecting the status-value - so the tree would be in loading state forever.
 Last time I looked into this the statement 

> loading if the query is in a "hard" loading state. This means there is no cached data and the query is currently fetching, eg isFetching === true

was not true, and it would be `loading` even if `fetchStatus: idle` in the case of disabled queries. 


We could probably improve this by checking the `fetchStatus` as well, eg. something like `loading: fetchStatus !== 'idle' && status === 'loading'`. However in this case I thought that blocking the UI when loading orgUnits isn't really that bad, but when I think more about it now - we probably shouldn't do that.

Reply:
> I see they've now also updated their v4 docs to be more accurate: loading: if there's no cached data and no query attempt was finished yet.

Yeah, saw that as well!

This has been updated and fixed in #94 . So we check for the data, as you suggested @ismay .

https://github.com/dhis2/data-entry-app/pull/94/files